File: part1-exercises-for-rust/hardways/leetcode-binary-tree-inorder-traversal/solution.c
int i=0;

struct TreeNode {
   int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

void inorder(struct TreeNode* root,int* array){
    if(!root) return;
    inorder(root->left,array);
    array[i++] = root->val;
    inorder(root->right,array);
}

int* inorderTraversal(struct TreeNode* root, int* returnSize){
    int *a = malloc(sizeof(int)*10000);
    i = 0;
    inorder(root,a);
    *returnSize = i;
    return a;
}

File: part1-exercises-for-rust/hardways/leetcode-binary-tree-inorder-traversal/solution.rs
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn inorder(root: &Option<Rc<RefCell<TreeNode>>>,mut array: Vec<i32>) -> Vec<i32> {
        if let Some(ref n) = root {
            array = Solution::inorder(&n.borrow().left,array);
            if let Some(ref v) = root{
                array.push(v.borrow().val);
            }
            array = Solution::inorder(&n.borrow().right,array);
        }
        return array;
    }

    pub fn inorder_traversal(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<i32> {
        let mut a = Vec::<i32>::new();
        Solution::inorder(&root,a)
    }
}